airflow-core/src/airflow/serialization/serialized_objects.py

```diff
@@ -1865,7 +1865,7 @@ def set_ref(task: Operator) -> Operator:
                 if _type == DAT.OP
                 else cls.deserialize_task_group(val, group, task_dict, dag=dag)
             )
-            for label, (_type, val) in encoded_group["children"].items()
+            for label, (_type, val) in sorted(encoded_group["children"].items(), key=lambda x: x[0])
```

Comment:
```suggestion
            for label, (_type, val) in sorted(encoded_group["children"].items())
```

#48534 will do this change first, with a test, but to fix test here I'm going to leave this in here.